### ISSUE TITLE
Add support for notification channels

### ISSUE THREAD

**Issue by dshokouhi** [home-assistant/android#107]
I know we don't have notifications yet but it would be great if we can add support for notification channels once we do. Its important to differentiate notifications so the user can set the priority and sound.  For example you don't want your phone making the same sound for the alarm being triggered as it does for the laundry.

https://developer.android.com/reference/android/app/NotificationChannel

**Comment by robbiet480** [comment]
My understanding is that notification channels are static and must be defined in AndroidManifest.xml which means we can't modify it during runtime. 

Firebase Cloud Messaging allows defining things like [sound](https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#AndroidNotification) and [priority](https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#androidmessagepriority) on a per notification basis.

**Comment by quthla** [comment]
Not sure if they can be defined in the manifest but they can defnitely be set programmatically.

**Comment by dshokouhi** [comment]
@quthla yup, my ring app does that as i add/remove ring devices.

**Comment by robbiet480** [comment]
My bad, I misread the FCM docs!

**Comment by RikP** [comment]
this will also support different messages being able to be send in do not disturb mode.
So same as critical messages on iphone.

**Comment by LordBoos** [comment]
Even static notification channels would be good. Like 3 channels for priorities like "low", "normal, "critical" so we can customize the channels with different sounds and settings in the android.
I'd like to set up critical notifications for things like "flooded bathroom" that ignore do not disturb mode but at the same time I don't want to low priority notifications like "plant needs water" to wake me up at night.

**Comment by dshokouhi** [comment]
Closed by #574 currently waiting on https://github.com/home-assistant/mobile-apps-fcm-push/pull/13 to be merged so the feature can work properly.  Once its working will get the docs updated.
